gr-blocks/python/blocks/qa_hier_block2.py

```diff
@@ -476,6 +476,32 @@ def thread_01(hblock, cls):
         hblock.wait()
         self.assertEqual(40, self.test_34b_val)
 
+    def test_035_lock_after_disconnect(self):
+        tb = gr.top_block()
+        src = blocks.vector_source_f([1.0, ])
+        hb = gr.hier_block2("hb",
+                            gr.io_signature(1, 1, gr.sizeof_float),
+                            gr.io_signature(1, 1, gr.sizeof_float))
+        m1 = multiply_const_ff(1.0)
+        hb.connect(hb, m1)
+        hb.connect(m1, hb)
+        dst1 = blocks.vector_sink_f()
+        dst2 = blocks.vector_sink_f()
+        tb.connect(src, hb, dst1)
+        tb.connect(hb, dst2)
+        tb.run()
+        tb.disconnect(hb, dst2)
+        hb.lock()
+        hb.unlock()
+        tb.run()
```

Comment:
This seems like the wrong order. A running flowgraph should be locked before it is modified, and unlocking it starts it running again. Also, it's sensible to wait for execution to finish before ending the test.
```suggestion
        tb.run()
        hb.lock()
        tb.disconnect(hb, dst2)
        hb.unlock()
        tb.wait()
```

Reply:
I've verified that the test still works properly after those changes; it segfaults on master and passes on this branch.

Reply:
The order is correct.
`tb.run()` calls `wait()`, so the flowgraph is not running at the time of `tb.disconnect()`
https://github.com/gnuradio/gnuradio/blob/main/gnuradio-runtime/lib/top_block.cc#L66-L70

Reply:
Oh, I was confusing `run()` and `start()`.

I don't think `lock()` does anything for a stopped flowgraph, so it might be a better test case if that was replaced with `start()`.